thavalon-webapp/src/components/Register.tsx

```diff
@@ -1,30 +1,86 @@
-import React, {useEffect, useState} from 'react';
+import React, {useState} from 'react';
 import ReactModal from 'react-modal';
-import { useForm } from 'react-hook-form';
-import "./modal.scss";
-import { register_user } from '../utils/account_utils';
+import { Resolver, useForm } from 'react-hook-form';
+import "./Modal.scss";
+import { AccountManager, HttpResponse } from '../utils/accountManager';
+import { Redirect } from 'react-router-dom';
 
+// Used by react modal for screen readers
 ReactModal.setAppElement("#root");
 
-function Login() {
+interface RegisterProps {
+    setLoggedIn: React.Dispatch<React.SetStateAction<boolean>>
+};
+
+interface RegisterData {
+    "name": string,
+    "email": string,
+    "password": string,
+    "confirmPassword": string,
+};
+
+
+const registerResolver: Resolver<RegisterData> = async (values: RegisterData) => {

```

Comment:
Could we use a non-exported function here instead of binding the function to a const?

Reply:
Doing that gives VSCode error:

Type '{ values: {}; errors: { confirmPassword: { type: string; message: string; }; } | {}; }' is not assignable to type 'Resolver<RegisterData, object>'.
  Object literal may only specify known properties, and 'values' does not exist in type 'Resolver<RegisterData, object>'.ts(2322)

Reply:
Since this doesn't use any Promises, does it need to be async?

Reply:
If I'm reading VSCode correctly, I think this Resolver is an alias for a promise:

(alias) type Resolver<TFieldValues extends Record<string, any> = Record<string, any>, TContext extends object = object> = (values: UnpackNestedValue<TFieldValues>, context?: TContext | undefined, validateAllFieldCriteria?: boolean | undefined) => Promise<ResolverResult<TFieldValues>> | ResolverResult<TFieldValues>

Reply:
Oh, I think the `=> Promise<ResolverResult> | ResolverResult` part means it can return either a `ResolverResult` or a `Promise<ResolverResult>`.

Does something like this work?

```
function registerResolver(values: RegisterData): ResolverResult {
  // ...
}
```